### ISSUE TITLE
Puppet run hangs when setting firewall rule

### ISSUE THREAD

**Issue by mike3085** [webalexeu/puppet-windows_firewall#9]
Trying to apply a firewall rule, the puppet run hangs indefinitely. Am on puppet 6.21.1 and puppetserver 6.15.1 - the client machines are Windows 10

```
   windows_firewall_rule { 'OpenSSH Server (sshd)':
        ensure          => present,
        direction       => 'inbound',
        action          => 'allow',
        enabled         => true,
        protocol        => 'tcp',
        local_port      => '22',
        remote_port     => 'any',
        display_name    => 'OpenSSH Server (sshd)',
        description     => 'Rule managed by Puppet'
    }
```

**Comment by webalexeu** [comment]
@mike406 
There is indeed a processing time issue when purge is not enabled and there is a lot of firewall rules defined on your system.
Can you confirm that is the case?
I'm currently working on a better solution to list the rules in order to solve the issue and I hope to publish a bugfix still for end of the week

**Comment by webalexeu** [comment]
@mike406 
Issue is now solved in v1.2.1
